File: Backend/src/main/java/com/bookstore/dto/AppleSignInRequest.java
package com.bookstore.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class AppleSignInRequest {
    /**
     * Apple identity token (JWT) from Sign in with Apple
     */
    @NotBlank(message = "Identity token is required")
    private String identityToken;

    /**
     * Authorization code for server-to-server verification (optional)
     */
    private String authorizationCode;

    /**
     * User's email (may be null after first login, or a private relay email)
     */
    private String email;

    /**
     * User's full name (combined given + family name, may be null after first login)
     */
    private String fullName;

    /**
     * Nonce for replay attack prevention
     */
    private String nonce;
}

File: App/lib/src/features/auth/data/models/apple_sign_in_request.dart
class AppleSignInRequest {
  final String identityToken;
  final String? authorizationCode;
  final String? email;
  final String? fullName;
  final String? nonce;

  AppleSignInRequest({
    required this.identityToken,
    this.authorizationCode,
    this.email,
    this.fullName,
    this.nonce,
  });

  Map<String, dynamic> toJson() {
    return {
      'identityToken': identityToken,
      if (authorizationCode != null) 'authorizationCode': authorizationCode,
      if (email != null) 'email': email,
      if (fullName != null) 'fullName': fullName,
      if (nonce != null) 'nonce': nonce,
    };
  }
}
